client/rest-client/src/main/java/org/infinispan/client/rest/impl/okhttp/RestResponseOkHttp.java

```diff
@@ -10,16 +10,19 @@
 import org.infinispan.commons.dataconversion.MediaType;
 
 import okhttp3.Response;
+import okhttp3.ResponseBody;
 
 /**
  * @author Tristan Tarrant &lt;tristan@infinispan.org&gt;
  * @since 10.0
  **/
 public class RestResponseOkHttp implements RestResponse {
    Response response;
+   private final ResponseBody body;
 
    RestResponseOkHttp(Response response) {
       this.response = response;
+      this.body = this.response.body();
```

Comment:
why caching the body? it seems to be simple field access.

Reply:
The body can be consumed only once, then it starts throwing errors...